test/integration/controller/jobs/mpijob/mpijob_controller_test.go

```diff
@@ -285,6 +287,183 @@ var _ = ginkgo.Describe("Job controller", ginkgo.Ordered, ginkgo.ContinueOnFailu
+				ResourceGroup(
+					*testing.MakeFlavorQuotas("test-flavor").Resource(corev1.ResourceCPU, "5").Obj(),
+				).AdmissionChecks("check").Obj()
+			gomega.Expect(k8sClient.Create(ctx, clusterQueueAc)).Should(gomega.Succeed())
+			localQueue = testing.MakeLocalQueue("queue", ns.Name).ClusterQueue(clusterQueueAc.Name).Obj()
+			gomega.Expect(k8sClient.Create(ctx, localQueue)).To(gomega.Succeed())
+			testFlavor = testing.MakeResourceFlavor("test-flavor").Label(instanceKey, "test-flavor").Obj()
+			gomega.Expect(k8sClient.Create(ctx, testFlavor)).Should(gomega.Succeed())
+			admissionCheck = testing.MakeAdmissionCheck("check").Obj()
+			gomega.Expect(k8sClient.Create(ctx, admissionCheck)).To(gomega.Succeed())
+
+			jobLookupKey = &types.NamespacedName{Name: jobName, Namespace: ns.Name}
+			wlLookupKey = &types.NamespacedName{Name: workloadmpijob.GetWorkloadNameForMPIJob(jobName), Namespace: ns.Name}
+		})
+
+		ginkgo.AfterEach(func() {
+			gomega.Expect(util.DeleteAdmissionCheck(ctx, k8sClient, admissionCheck)).To(gomega.Succeed())
+			util.ExpectResourceFlavorToBeDeleted(ctx, k8sClient, testFlavor, true)
+			gomega.Expect(util.DeleteNamespace(ctx, k8sClient, ns)).To(gomega.Succeed())
+			util.ExpectClusterQueueToBeDeleted(ctx, k8sClient, clusterQueueAc, true)
+		})
+
+		ginkgo.It("labels and annotations should be propagated from admission check to job", func() {
+			createdJob := &kubeflow.MPIJob{}
+			createdWorkload := &kueue.Workload{}
+
+			ginkgo.By("creating the job with pod labels & annotations", func() {
+				job := testingmpijob.MakeMPIJob(jobName, ns.Name).
+					Queue(localQueue.Name).
+					PodAnnotation(kubeflow.MPIReplicaTypeWorker, "old-ann-key", "old-ann-value").
+					PodLabel(kubeflow.MPIReplicaTypeWorker, "old-label-key", "old-label-value").
+					Obj()
+				gomega.Expect(k8sClient.Create(ctx, job)).Should(gomega.Succeed())
+			})
+
+			ginkgo.By("fetch the job and verify it is suspended as the checks are not ready", func() {
+				gomega.Eventually(func() *bool {
+					gomega.Expect(k8sClient.Get(ctx, *jobLookupKey, createdJob)).Should(gomega.Succeed())
+					return createdJob.Spec.RunPolicy.Suspend
+				}, util.Timeout, util.Interval).Should(gomega.Equal(ptr.To(true)))
+			})
+
+			ginkgo.By("fetch the created workload", func() {
+				gomega.Eventually(func() error {
+					return k8sClient.Get(ctx, *wlLookupKey, createdWorkload)
+				}, util.Timeout, util.Interval).Should(gomega.Succeed())
+			})
+
+			ginkgo.By("add labels & annotations to the admission check", func() {
+				gomega.Eventually(func() error {
+					var newWL kueue.Workload
+					gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(createdWorkload), &newWL)).To(gomega.Succeed())
+					workload.SetAdmissionCheckState(&newWL.Status.AdmissionChecks, kueue.AdmissionCheckState{
+						Name:  "check",
+						State: kueue.CheckStateReady,
+						PodSetUpdates: []kueue.PodSetUpdate{
+							{
+								Name: "launcher",
```

Comment:
same, maybe add some updates for this podset

Reply:
sure, done